tokio-trace/src/instrument.rs

```diff
@@ -24,10 +24,18 @@ impl<T: Future> Future for Instrumented<T> {
     type Error = T::Error;
 
     fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
-        let span = &self.span;
+        let span = self.span.take()
+            .expect("polled after Ready");
```

Comment:
Nit: if this is an invariant that can't be broken, I think I'd prefer that the panic message was a bit more wordy about the cause.

Reply:
yeah... in this case I'm about 95% certain this code _cannot_ enter a s state where this will panic, but you're not wrong